src/sentry/api/endpoints/organization_artifactbundle_assemble.py

```diff
@@ -147,7 +151,19 @@ def post(self, request: Request, organization) -> Response:
             organization_id=organization.id,
             project_ids=project_ids,
             user_agent=request.META.get("HTTP_USER_AGENT", ""),
-            auth_type="api_token" if is_api_token_auth(request.auth) else None,
+            auth_type="api_token"
+            if is_api_token_auth(request.auth)
+            else "org_auth_token"
+            if is_org_auth_token_auth(request.auth)
+            else None,
         )
 
+        if is_org_auth_token_auth(request.auth):
+            org_auth_token_id = get_org_auth_token_id_from_auth(request.auth)
+            orgauthtoken_service.update_org_auth_token(
+                org_auth_token_id=org_auth_token_id,
+                date_last_used=datetime.utcnow(),
+                project_last_used_id=project_ids[0] if len(project_ids) > 0 else None,
+            )
```

Comment:
Is there any way you could collapse such behavior into a function? To me it seems that there is too much repetition across the board and lots of params are common across contexts.

```python
def update_last_used_project(auth, project_ids):
	if is_org_auth_token_auth(auth):
	          org_auth_token_id = get_org_auth_token_id_from_auth(auth)
	          orgauthtoken_service.update_org_auth_token(
	              org_auth_token_id=org_auth_token_id,
	              date_last_used=datetime.utcnow(),
	              project_last_used_id=project_ids[0] if len(project_ids) > 0 else None,
	          )
```

What do you think?

Reply:
yeah, good point - I actually initially had it like this, then moved stuff to the HC service, but we can still wrap this away 👍